### PR TITLE
Renamed FusedOps to TrinaryOps and added TrinaryOps.WHERE

### LINKED ISSUE #1154: Add WHERE TrinaryOp/mlop
Rename FusedOps to TrinaryOps.

The solution for #942

### PR DESCRIPTION
**Pull request by flammit**
This is a draft PR pending the resolution of the Trinary vs Ternary naming as well as the broadcasting code refactoring.

Closes #1154 and #942

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/codegen/assembly_ptx.py           2               2
tinygrad/codegen/assembly_rdna.py          3               3
tinygrad/codegen/cstyle.py                 4               2
tinygrad/codegen/linearizer.py             7               6
tinygrad/codegen/llvmir.py                 4               2
tinygrad/graph.py                          3               3
tinygrad/lazy.py                           2               1
tinygrad/mlops.py                         14               1
tinygrad/ops.py                            7               6
tinygrad/runtime/ops_cpu.py                3               2
tinygrad/runtime/ops_metal.py              1               1
tinygrad/runtime/ops_torch.py              3               2
tinygrad/tensor.py                        26               5
------------------------------------------------------------
total                                     79              36
------------------------------------------------------------
lines added in the tinygrad folder: 43
```

File: tinygrad/runtime/ops_metal.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 class _METAL:
   def __init__(self):
     self.mtl_buffers_in_flight: List[Any] = []
-    self.device = Metal.MTLCreateSystemDefaultDevice()
+    self.device = [x for x in Metal.MTLCopyAllDevices() if x.name() not in getenv('CL_EXCLUDE', '').split(',')][0]
```
Comment (wozeparrot) [inline tinygrad/runtime/ops_metal.py:15]: Why was this added?

Comment (flammit) [reply tinygrad/runtime/ops_metal.py:15]: Whoops - I will remove this and fix the linter issues.

Comment (flammit) [reply tinygrad/runtime/ops_metal.py]: Apologies - I was testing other METAL code and needed to exclude the Intel GPU (defaults to it).  I've reverted this in the latest commit.

**Comment by geohot** [comment]
Can you work with #1196 to make one great PR?

**Comment by flammit** [comment]
> Can you work with #1196 to make one great PR?

Will do! Looks like the naming will end up ternary anyway so I’ll push a PR to #1196 with my suggested changes and close this one when the poll is complete.

**Comment by geohot** [comment]
The verdict is `TernaryOps` btw. I like this PR marginally more, the changes to broadcasted in #1196 will make it slow.

**Comment by flammit** [comment]
Closing as dup to #1196
